pontoon/sync/changeset.py

```diff
@@ -234,25 +238,30 @@ def update_entity_translations_from_vcs(
                 ))
 
         # Any existing translations that were not approved get unapproved.
-        if old_translations is None:
-            old_translations = db_translations.filter(approved_date__lte=self.now)
-
         for translation in old_translations:
             if translation not in approved_translations:
+                # Use the translation instance already set for update if it exists
+                translation = (
+                    match_attr(self.translations_to_update, pk=translation.pk) or translation
```

Comment:
NIT: Set is nice, but dict could reduce that line into `self.tranlations_to_update.get(translation.pk, translation)` and _probably_ could be faster than match_attr function but _relatively_ less explicit. Still, the PR lgtfm :+1: